test/characterization/generate_data.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Generate data for characterization fitters tests
+"""
+
+import os
+import pickle
+
+from qiskit.ignis.characterization.coherence.fitters import T1Fitter, \
+                                                            T2Fitter, \
+                                                            T2StarFitter
+
+from qiskit.ignis.characterization.hamiltonian.fitters import ZZFitter
+
+
+def generate_data(filename_prefix):
+    """
+    Generate pickle files
+    """
+
+    files_with_pickles = ['test_fitters_t1.pkl', 'test_fitters_t2.pkl',
+                          'test_fitters_t2star.pkl',
+                          'test_fitters_zz.pkl']
+
+    for picfile in files_with_pickles:
+        fo = open(os.path.join(os.path.dirname(__file__),
+                               picfile), 'rb')
+        file_content = pickle.load(fo)
+        fo.close()
+
+        fit_type = file_content['type']
+
+        input_to_fitter = file_content['input_to_fitter']
```

Comment:
What is the`input_to_fitter` data? This looks like it is loading existing pickled data and saving to a new pickle file, but doesn't have anything to generate new data from scratch (ie running a simulation).

If the input reference data is just counts, can they be saved in a `.py` or JSON file, and then we don't need pickles at all?

Reply:
The pickles will still be needed to check the fit results

Reply:
The fitters take as input the entire backend result. I don't remember why all this is needed, and probably can be changed to take only the counts. So the input to the fitters is very long therefore has to be stored in a file, either in pickle or in json, if json agrees. I prefer the way it is now, with input and output stored in the same file. The alternative is a file for input and a file for output.